packages/toolkit/src/lib/react-query-service/catalog/useListCatalogFiles.ts

```diff
@@ -1,98 +1,39 @@
 import { useQuery } from "@tanstack/react-query";
-import { getQueryString, Nullable } from "instill-sdk";
+import { File, Nullable } from "instill-sdk";
 
-import { createInstillAxiosClient } from "../../sdk-helper";
-import { File } from "./types";
-
-export async function listCatalogFiles({
-  namespaceId,
-  catalogId,
-  accessToken,
-  pageSize,
-  pageToken,
-}: {
-  namespaceId: string;
-  catalogId: string;
-  accessToken: string;
-  pageSize: number;
-  pageToken: Nullable<string>;
-}): Promise<{
-  files: File[];
-  pageToken: Nullable<string>;
-  totalSize: number;
-}> {
-  const client = createInstillAxiosClient(accessToken, true);
-  const queryString = getQueryString({
-    baseURL: `/namespaces/${namespaceId}/catalogs/${catalogId}/files`,
-    pageSize,
-    pageToken: pageToken ?? undefined,
-  });
-  try {
-    const { data } = await client.get<{
-      files: File[];
-      pageToken: Nullable<string>;
-      totalSize: number;
-    }>(queryString);
-    return {
-      files: data.files,
-      pageToken: data.pageToken,
-      totalSize: data.totalSize,
-    };
-  } catch (error) {
-    return Promise.reject(error);
-  }
-}
+import { getInstillCatalogAPIClient } from "../../sdk-helper";
 
 export function useListCatalogFiles({
   namespaceId,
   catalogId,
   accessToken,
   enabled,
 }: {
-  namespaceId: Nullable<string> | undefined;
+  namespaceId: Nullable<string>;
   catalogId: Nullable<string>;
   accessToken: Nullable<string>;
   enabled: boolean;
 }) {
   return useQuery<File[]>({
     queryKey: ["catalogFiles", namespaceId, catalogId],
     queryFn: async () => {
-      if (!accessToken) {
-        throw new Error("accessToken not provided");
-      }
-      if (!namespaceId) {
-        throw new Error("namespaceId not provided");
-      }
-      if (!catalogId) {
-        throw new Error("catalogId not provided");
+      if (!namespaceId || !accessToken || !catalogId) {
```

Comment:
Do we want to generalise here this error?
Yeah, I understand that most likely we'll never see it, but still.
In case it throws we won't know which parameter exactly was missing.